### DISCUSSION TITLE
FGSM on SKlearn models

### DISCUSSION THREAD

**Discussion by MaSe1301** [Trusted-AI/adversarial-robustness-toolbox#1539] [Q&A] [answered]
Hello

Thanks for this great toolbox:)

For a student research project I would like to test if ensemble methods like blending can improve the adversial robustness. As a base learner I use different Sklearn models. As whitebox attack I use the FastGradient attack.
I could test the logistic regression this way, but with other methods like Decision Tree, Stochastic Gradiend Descendend I get the following error:
![grafik](https://user-images.githubusercontent.com/86280970/153481252-5ff951f3-e8b0-4b97-983d-d2159bbb2950.png)

Is there a way to solve this error?
or exists a whitebox attack that works for the following classifiers:
Logistic Regression (sklearn)
LinearDiscriminantAnalysis (sklearn)
Stochastic Gradient Descended (sklearn)
Decision Tree (sklearn)
Gaussian naive bayes (sklearn)
Convonutional neural network (keras)

Thanks for the help

**Comment by beat-buesser** [comment, marked as answer]
Hi @MaSe1301 Thank you for using ART! This exception means that the specific combination of estimator and attack is not possible because of the requirements of the attack. In your case specifically it is because FastGradient requires loss gradients and a decision tree based classifiers cannot calculate loss gradients.

**Reply by MaSe1301** [reply]
Hey

Thanks for the answer.)

I am relatively new to machine learning and I am sorry if the question is stupid. Is there an white box adversial attack that fits all sklearn learners? Term of my thesis is to show , that blending can improve reliability and robustness. I already generate some fault injections with blur, noise..., but the result was not so convincing.
can you recommend a procedure to test the robustness?

Thx for the help!

**Reply by beat-buesser** [reply]
I don't think there is one white-box attack that fits all possible models because the white-box scenarios exploits specific aspects of a model. You will have to choose an attack the is compatible with a specific model type.
